server/src/main/java/com/flipkart/varadhi/web/v1/authz/AuthZHandlers.java

```diff
@@ -0,0 +1,205 @@
+                                HttpMethod.GET,
+                                "/projects/:project/policy",
+                                Set.of(authenticated),
+                                new LinkedHashSet<>(),
+                                this.getIAMPolicyHandler(ResourceType.PROJECT),
+                                true,
+                                Optional.empty()
+                        ),
+                        new RouteDefinition(
+                                HttpMethod.PUT,
+                                "/projects/:project/policy",
+                                Set.of(hasBody, authenticated),
+                                new LinkedHashSet<>(),
+                                this.setIAMPolicyHandler(ResourceType.PROJECT),
+                                true,
+                                Optional.empty()
+                        ),
+                        new RouteDefinition(
+                                HttpMethod.GET,
+                                "/projects/:project/topics/:topic/policy",
+                                Set.of(authenticated),
+                                new LinkedHashSet<>(),
+                                this.getIAMPolicyHandler(ResourceType.TOPIC),
+                                true,
+                                Optional.empty()
+                        ),
+                        new RouteDefinition(
+                                HttpMethod.PUT,
+                                "/projects/:project/topics/:topic/policy",
+                                Set.of(hasBody, authenticated),
+                                new LinkedHashSet<>(),
+                                this.setIAMPolicyHandler(ResourceType.TOPIC),
+                                true,
+                                Optional.empty()
+                        )
+                )
+        ).get();
+    }
+
+    @Override
+    public List<RouteDefinition> get() {
+        return Stream.of(
+                getDebugHandlers(),
+                getPolicyHandlers()
+        ).flatMap(List::stream).toList();
+    }
+
+    public Handler<RoutingContext> getIAMPolicyHandler(ResourceType resourceType) {
+        return (routingContext) -> {
+            String resourceId = routingContext.getResourceIdFromPath(resourceType);
+            RoleBindingNode policy = getIAMPolicy(resourceType, resourceId);
+            routingContext.endApiWithResponse(policy);
+        };
+    }
+
+    public RoleBindingNode getIAMPolicy(ResourceType resourceType, String resourceId) {
+        return authZService.getIAMPolicy(resourceType, resourceId);
+    }
+
+    public Handler<RoutingContext> setIAMPolicyHandler(ResourceType resourceType) {
+        return (routingContext) -> {
+            String resourceId = routingContext.getResourceIdFromPath(resourceType);
+            IAMPolicyRequest policyForSubject = routingContext.body().asPojo(IAMPolicyRequest.class);
```

Comment:
IAMPolicyRequest looks like can go to entities module.
entities.auth package.

Reply:
done